### DISCUSSION TITLE
"Compatibility" rebirth ramifications

### DISCUSSION THREAD

**Discussion by JeodC** [dxx-rebirth/dxx-rebirth#749] [General]
Last week I created a compatibility branch on forked dxx-rebirth and modified `valptridx.h` in commit https://github.com/JeodC/dxx-rebirth/commit/c22de974133f407e6413dafde8c0769019881fec. This allows the port to run on operating systems that use older GLIBC (and compiles on bullseye).

I never found out what the ramifications for this are. Since I didn't build from this repo, I assume git headers and hashes are different. Does this mean my port's multiplayer will be restricted? If so, are there any workarounds I can do?

Are there any other possible ramifications I could encounter by using the above commit?

Thanks in advance. @vLKp

**Comment by JeodC** [comment]
It seems I can connect my retro handheld to my Windows build which does not have the compatiblity adjustment. This is good news.

**Comment by vLKp** [comment]
Compatibility is enforced via the build version number (`DXX_VERSION_MAJORi`, `DXX_VERSION_MINORi`, `DXX_VERSION_MICROi`) and the multiplayer protocol version number (`MULTI_PROTO_VERSION`).  If you keep those the same, your build will try to play with stock builds.  However, this is a double-edged sword: if you change the multiplayer protocol ABI (renumbering messages, adding/removing fields, etc.) and do _not_ change any of those 4 variables, then the resulting build will still try to play with stock builds, but probably will not work since they are not speaking the same protocol.  Consequences could vary from the obvious, like failing to join a game, to the subtle and bizarre, like objects being in the wrong place because the clients disagreed about how to describe the object's location to each other.  In practice, I try not to change the multiplayer protocol if avoidable, and I do update those variables when I change it.  Therefore, the most likely result is that everything is fine.  The second most likely result is that you try to cherry-pick one of my changes into your fork, and it fails to `git cherry-pick` because it depends on a prior commit that you did not cherry-pick.  This will be very obvious to you, and will stop further work while you sort out how to guide `git` through the merge that it could not do on its own.

---

The git commit will be different since you used `git commit` to save your changes relative to `master`.  This will cause the output of `git describe` to be different, so the version embedded in `g_descent_version` (the contents of which are shown on the main menu, among other places) will not match what dxx-rebirth.git's `master` would build.  This is effectively cosmetic, but it does mean that if someone posts asking for help with your port, I won't be able to map their build to a commit in dxx-rebirth.git because it is not built from dxx-rebirth.git, and their output will refer to a commit from your fork.  This will be mildly troublesome, since it means I cannot readily tell how many commits from dxx-rebirth.git's `master` they lack, nor tell how many commits of yours they have that are not in `master`.  Since you are mostly committing workarounds for outdated compilers, not knowing how many commits you have done is probably fine.  I would be more concerned about that angle if you were actively developing your fork in a different direction, adding features and bugs different from Rebirth.  Not being able to determine what fixes from Rebirth they lack will be a more direct problem, but as long as your history is published, I can probably use their version information to resolve it.  It will just be less convenient than a simple `git log`.

**Comment by JeodC** [comment]
There is currently no git describe on the port's menu screen, presumably because it's forked and scons couldn't grab the info it wanted. Like I said before, though, I don't believe maintaining compatibility for an outdated custom firmware is in my best interest, nor yours. If I do create future builds for debian arm64, I will end support for ArkOS plain and simple.
